vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1259 @@
+            int name = tile.getName();
+            loc_ loc = tile_to_loc_[name];
+            const auto& tile_type = ar_.getTileTypeList()[tile.getType()];
+
+            for (const auto& pip : tile_type.getPips()) {
+                if (pip.isPseudoCells())
+                    continue;
+                if (tile.hasSubTilesPrefices())
+                    name = tile.getSubTilesPrefices()[pip.getSubTile()];
+                int wire0_name, wire1_name;
+                int node0, node1;
+                int switch_id;
+
+                wire0_name = tile_type.getWires()[pip.getWire0()];
+                wire1_name = tile_type.getWires()[pip.getWire1()];
+                if (wire_to_node_.find(std::tuple<int, int>(name, wire0_name)) == wire_to_node_.end())
+                    continue;
+                if (wire_to_node_.find(std::tuple<int, int>(name, wire1_name)) == wire_to_node_.end())
+                    continue;
+                node0 = wire_to_node_[std::tuple<int, int>(name, wire0_name)];
+                node1 = wire_to_node_[std::tuple<int, int>(name, wire1_name)];
+
+                used_by_pip_.insert(std::tuple<int, loc_>(node0, loc));
+                used_by_pip_.insert(std::tuple<int, loc_>(node1, loc));
+                std::tuple<int, int> source_sink(node0, node1);
+                switch_id = pips_models_[std::tuple<int, bool>(pip.getTiming(), pip.getBuffered21())];
+                pips_[source_sink] = std::tuple<int, std::tuple<int, int, int, bool>>{switch_id, {name, wire0_name, wire1_name, true}};
+                if (!pip.getBuffered21()) {
+                    source_sink = std::tuple<int, int>(node1, node0);
+                    pips_[source_sink] = std::tuple<int, std::tuple<int, int, int, bool>>{switch_id, {name, wire0_name, wire1_name, true}};
+                }
+                if (!pip.getDirectional()) {
+                    source_sink = std::tuple<int, int>(node1, node0);
+                    switch_id = pips_models_[std::tuple<int, bool>(pip.getTiming(), pip.getBuffered20())];
+                    pips_[source_sink] = std::tuple<int, std::tuple<int, int, int, bool>>{switch_id, {name, wire0_name, wire1_name, false}};
+                    if (!pip.getBuffered20()) {
+                        source_sink = std::tuple<int, int>(node0, node1);
+                        pips_[source_sink] = std::tuple<int, std::tuple<int, int, int, bool>>{switch_id, {name, wire0_name, wire1_name, true}};
+                    }
+                }
+            }
+        }
+    }
+
+    loc_ offsets[4] = {loc_(-1, 0), loc_(0, 1), loc_(1, 0), loc_(0, -1)};
+
+    /*
+     * Build graph of FPGA Interchange node for further computations
+     */
+
+    node_* build_node_graph(int node_id,
+                            std::set<loc_> nodes,
+                            std::set<loc_> empties,
+                            int root_x,
+                            int root_y) {
+        int cnt = 0;
```

Comment:
I am not sure how this `cnt` represents actually, some description around it may be useful, and possibly also rename it to something more descriptive.